tokio/src/sync/broadcast.rs

```diff
@@ -893,20 +894,52 @@ impl<T> Shared<T> {
         //   guard node after this function returns / panics.
         let mut list = WaitersList::new(std::mem::take(&mut tail.waiters), guard.as_ref(), self);
 
+        // From now on, read lock suffices: we own our own copy of waiters list,
+        // and we only need to guard against concurrent waiter removals.
+        // Except us, waiter removals are done by `Recv::drop` and it takes
+        // a write lock to do it.
+        /*
+        let mut tail = if cfg!(all(
```

Comment:
Also I got lost in what are the right feature flags to use here to make all the builds work, so I dropped it for now.

Reply:
Is there such a wrapper? I found one [here](https://github.com/tokio-rs/tokio/blob/master/tokio/src/loom/std/parking_lot.rs#L43), but it wraps the `parking_lot` guard, not the std one. For the std case, it looks like std types are [just reexported](https://github.com/tokio-rs/tokio/blob/master/tokio/src/loom/std/mod.rs#L67).

Ok now I see that there is a wrapper for Mutex, I can create the same wrapper for RwLock.